olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -190,13 +227,97 @@ class AuthenticationClient::Impl final {
       const AuthenticationCredentials& credentials,
       const std::string& userAccessToken, const SignOutUserCallback& callback);
 
+  client::CancellationToken IntrospectApp(std::string access_token,
+                                          IntrospectAppCallback callback);
+
   void SetNetworkProxySettings(
       const http::NetworkProxySettings& proxy_settings);
 
   void SetNetwork(std::shared_ptr<http::Network> network);
 
   void SetTaskScheduler(std::shared_ptr<TaskScheduler> task_scheduler);
 
+  static olp::authentication::IntrospectAppResult parseIntrospectAppResult(
+      const std::shared_ptr<rapidjson::Document>& doc) {
```

Comment:
If you do not need to copy or work on the shared_ptr then pass instead reference to rapidjson::Document or to const rapidjson::Document.

```suggestion
      const rapidjson::Document& doc) {
```